src/client/Client.cc

```diff
@@ -8707,7 +8707,7 @@ int Client::preadv(int fd, const struct iovec *iov, int iovcnt, loff_t offset)
   return _preadv_pwritev(fd, iov, iovcnt, offset, false);
 }
 
-int Client::_read(Fh *f, int64_t offset, uint64_t size, bufferlist *bl)
+int64_t Client::_read(Fh *f, int64_t offset, uint64_t size, bufferlist *bl)
```

Comment:
I'm a bit confused by what you mean about converting int64->int "at a higher level".
eg, ::_read() is called by ::read() up at line 8694, and that's stuffing the _read() result into an int. Did you just miss one, or did you mean for that to be the conversion point?

If you're worried about integer overflows, shouldn't there be size checks going in on or something? I don't really understand what it has to do with preadv/pwritev support.

Reply:
No, my intent was to not change how the existing functions that return an int work. They still cast the result to an int for now, but the underlying layer that ll_readv/ll_writev now call should be 64-bit clean.

...and yeah, I think we probably do want to do size checking better than we do now, but we have API functions like this:

```
int ceph_write(struct ceph_mount_info *cmount, int fd, const char *buf, int64_t size, int64_t offset);
```

What should we do when someone gives a "size" value that is larger than INT_MAX ? I think we probably want to clamp the I/O size to that, right?

Reply:
>  I think we probably want to clamp the I/O size to INT_MAX, right?

Sounds reasonable to me.

Reply:
Ok, I'll see if I can add another patch to clamp the I/O sizes to INT_MAX in those functions.